src/components/TransactionImporter.js

```diff
@@ -33,22 +37,27 @@ export default class TransactionImporter extends React.Component {
     return <div className="xdrInput">
       <div className="xdrInput__input">
         <textarea
+          value={this.state.input}
           className="xdrInput__input__textarea"
           onChange={this.updateTextarea.bind(this)}
           placeholder="Example: AAAAAGXNhB2hIkbP//jgzn4os/AAAAZAB+BaLPAAA5Q/xL..."></textarea>
+      
       </div>
       <div className="xdrInput__message">
         {message}
       </div>
+      {this.props.onImport &&
       <div className="s-buttonList">
         <button className="s-button"
           disabled={!submitEnabled} onClick={this.triggerImport.bind(this)}>
           Import Transaction</button>
       </div>
+      }
     </div>
   }
 }
 TransactionImporter.propTypes = {
-  onImport: PropTypes.func.isRequired,
+  onImport: PropTypes.func,
   networkPassphrase: PropTypes.string.isRequired,
+  onUpdate: PropTypes.func,
 };
```

Comment:
allowing TransactionImporter to be used as an XDR input w/ validation w/o having to use the import button